packages/form-js-viewer/src/render/components/form-fields/Table.js

```diff
@@ -1,36 +1,405 @@
-const type = 'table';
+import { isDefined, isNumber, isObject, isString } from 'min-dash';
+import { useExpressionEvaluation } from '../../hooks';
+import { useEffect, useState } from 'preact/hooks';
+import Label from '../Label';
+import { formFieldClasses, prefixId } from '../Util';
+import ArrowDownIcon from './icons/ArrowDown.svg';
+import ArrowUpIcon from './icons/ArrowUp.svg';
+import CaretLeftIcon from './icons/CaretLeft.svg';
+import CaretRightIcon from './icons/CaretRight.svg';
+import classNames from 'classnames';
 
+const type = 'table';
 
+/**
+ * @typedef {('asc'|'desc')} Direction
+ *
+ * @typedef Sorting
+ * @property {string} key
+ * @property {Direction} direction
+ *
+ * @typedef Column
+ * @property {string} label
+ * @property {string} key
+ *
+ * @typedef Props
+ * @property {boolean} disabled
+ * @property {Object} field
+ * @property {string} field.id
+ * @property {Array<Column>|void} field.columns
+ * @property {string|void} field.columnsExpression
+ * @property {string|void} field.label
+ * @property {number|void} field.rowCount
+ * @property {string|void} field.dataSource
+ *
+ * @param {Props} props
+ * @returns {import("preact").JSX.Element}
+ */
 export default function Table(props) {
-  return 'Table';
+  const { field, disabled } = props;
+  const {
+    columns = [],
+    columnsExpression,
+    dataSource = '',
+    rowCount,
+    id,
+    label,
+  } = field;
+
+  /** @type {[(null|Sorting), import("preact/hooks").StateUpdater<null|Sorting>]} */
+  const [ sortBy, setSortBy ] = useState(null);
+  const evaluatedColumns = useEvaluatedColumns(
+    columnsExpression || '',
+    columns,
+  );
+  const columnKeys = evaluatedColumns.map(({ key }) => key);
+  const evaluatedDataSource = useExpressionEvaluation(
+    dataSource.startsWith('=') ? dataSource : `=${dataSource}`,
+  );
+  const data = Array.isArray(evaluatedDataSource) ? evaluatedDataSource : [];
+  const sortedData =
+    sortBy === null
+      ? data
+      : sortByColumn(data, sortBy.key, sortBy.direction);
+
+  /** @type {unknown[][]} */
+  const chunkedData = isNumber(rowCount) ? chunk(sortedData, rowCount) : [ sortedData ];
+  const [ currentPage, setCurrentPage ] = useState(0);
+  const currentChunk = chunkedData[currentPage] || [];
+
+
+  useEffect(() => {
+    setCurrentPage(0);
+  }, [ rowCount, sortBy ]);
+
+  if (disabled) {
+    return <DisabledTable { ...props } />;
```

Comment:
Disabled is an actual state we use (although deprecated). The proper way to create behavior unique to the table in the editor would be to create an EditorTable, this is more hijacking an existing concept for a different purpose which I'm not 100% fond of. See `EditorText` for reference, it's pretty simple to change though and would move some non-viewer code outside of the viewer project.